docs/API.md

```diff
@@ -6,6 +6,10 @@ These are best practices related to the current API used for App. This does not
 - There should be no client-side handling that is unique to any API call.
 - Data is pushed to the client and put straight into Onyx by low-level libraries.
 - Clients should be kept up-to-date with many small incremental changes to data.
+- Creating data needs to be flawlessly fast on every connection (offline, slow 3G, etc).
+- For new objects created from the client (reports, reportActions, policies) we're going to generate a random string ID immediately on the client, rather than needing to wait for the server to give us an ID for the created object.
```

Comment:
https://expensify.slack.com/archives/C02HWMSMZEC/p1655438492403349?thread_ts=1654882263.720009&cid=C02HWMSMZEC

Reply:
You can also add:

> This client-generated ID will become the primary key for that record in the database. This will provide more offline functionality than was previously possible.

Just so it's clear what happens with the ID.

Reply:
Cool, added as a sub-bullet point